gmp-droid.cpp

```diff
@@ -0,0 +1,559 @@
+/****************************************************************************
+**
+** Copyright (c) 2020 Open Mobile Platform LLC.
+**
+** This Source Code Form is subject to the terms of the
+** Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+** with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+**
+****************************************************************************/
+
+#include <iostream>
+#include <cstring>
+#include <map>
+#include <stdlib.h>
+#include <arpa/inet.h>
+
+#include "droidmediacodec.h"
+
+#include "gmp-platform.h"
+#include "gmp-video-host.h"
+
+#include "gmp-video-decode.h"
+#include "gmp-video-frame-i420.h"
+#include "gmp-video-frame-encoded.h"
+#include "gmp-droid-conv.h"
+#include "gmp-task-utils.h"
+
+#define CRITICAL 0
+#define ERROR 1
+#define INFO  2
+#define DEBUG 3
+
+const char *kLogStrings[] = {
+  "GMP-DROID Critical: ",
+  "GMP-DROID Error: ",
+  "GMP-DROID Info: ",
+  "GMP-DROID Debug: "
+};
+
+static int g_log_level = INFO;
+
+#define LOG(l, x) do { \
+        if (l >=0 && l <= g_log_level) { \
+            std::cerr << kLogStrings[l] << x << std::endl; \
+        } \
+    } while(0)
+
+static GMPPlatformAPI *g_platform_api = NULL;
+
+// Droidmedia callbacks
+static void DroidError (void *data, int err);   // pass error and abord
+static int SizeChanged (void *data, int32_t width, int32_t height);     // reconfigure
+static void DataAvailable (void *data, DroidMediaCodecData * decoded);
+static void SignalEOS (void *data);
+
+class DroidVideoDecoder : public GMPVideoDecoder
+{
+public:
+  explicit DroidVideoDecoder (GMPVideoHost * hostAPI):
+      m_host (hostAPI),
+      m_callback (NULL)
```

Comment:
Having this like:
```
: m_host (hostAPI)
, m_callback (nullptr)
```
Would cause few changes when more members are added.

Reply:
+
```
, m_codec_lock (nullptr)
, m_stop_lock (nullptr)
```

Reply:
Done